pkg/sql/tpp/parser.go

```diff
@@ -91,3 +86,50 @@ func tiDBParseAndSplit(sql string) ([]string, int, error) {
 	}
 	return sqls, -1, nil
 }
+
+type parseResult struct {
+	Statements []string `json:"statements"`
+	Position   int      `json:"position"`
+	Error      string   `json:"error"`
+}
+
+func javaParseAndSplit(typ, sql string) ([]string, int, error) {
+	// cwd is used to store train scripts and save output models.
+	cwd, err := ioutil.TempDir("/tmp", "sqlflow")
+	if err != nil {
+		return nil, -1, err
+	}
+	defer os.RemoveAll(cwd)
+
+	inputFile := filepath.Join(cwd, "input.sql")
+	outputFile := filepath.Join(cwd, "output.json")
+	if err := ioutil.WriteFile(inputFile, []byte(sql), 755); err != nil {
+		return nil, -1, err
+	}
+
+	cmd := exec.Command("java",
+		"-cp", "/opt/sqlflow/parser/parser-1.0-SNAPSHOT-jar-with-dependencies.jar",
+		"org.sqlflow.parser.ParserAdaptorCmd",
+		"-p", typ,
+		"-i", inputFile,
```

Comment:
Maybe we can refine this later using the Linux pipe to avoid writing temporary files on the disk.